superset-frontend/src/dashboard/components/nativeFilters/FilterBar/Vertical.tsx

```diff
@@ -220,7 +220,12 @@ const VerticalFilterBar: React.FC<VerticalBarProps> = ({
           />
         </FilterControlsWrapper>
       ),
-    [canEdit, dataMaskSelected, filterValues.length, onSelectionChange],
+    [
+      canEdit,
+      JSON.stringify(dataMaskSelected),
```

Comment:
Please fix the useFilterDependencies missing shallowEqual rather than this change.

https://github.com/apache/superset/blob/0c083bdc1af4e6a3e17155246a3134cb5cb5887d/superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/state.ts#L33-L35

```
import { shallowEqual, useSelector } from 'react-redux';
...
  const dependencyIds = useSelector<any, string[] | undefined>(
    state => state.nativeFilters.filters[id]?.cascadeParentIds,
    shallowEqual, // <-
  );
```

```suggestion
      dataMaskSelected,
```